firstvoices/backend/views/custom_order_recalculate_views.py

```diff
@@ -0,0 +1,81 @@
+import rules
+from celery.result import AsyncResult
+from django.core.exceptions import PermissionDenied
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from backend.models import CustomOrderRecalculationPreviewResult, Site
+from backend.tasks.alphabet_tasks import recalculate_custom_order_preview
+
+
+class CustomOrderRecalculatePreviewView(APIView):
+    task_id = None
+
+    @staticmethod
+    def check_superadmin_status(request):
+        if not rules.has_perm("views.has_custom_order_access", request.user):
```

Comment:
This is checking a permission, so please call it something like check_permission